libs/types-rs/src/geometry.rs

```diff
@@ -280,95 +394,42 @@ pub enum IntersectionBounds {
     Bounded,
 }
 
-/// Determines an intersection point of two line segments. If `bounded` is set
-/// to `true`, the intersection point must be within the bounds of both
-/// segments. If `bounded` is set to `false`, the intersection point may be
-/// outside the bounds of either segment.
-#[must_use]
-pub fn intersection_of_lines(
-    segment1: &Segment,
-    segment2: &Segment,
-    bounds: IntersectionBounds,
-) -> Option<Point<SubPixelUnit>> {
-    let p1 = segment1.start;
-    let p2 = segment1.end;
-    let p3 = segment2.start;
-    let p4 = segment2.end;
-    let d = (p4.y - p3.y).mul_add(p2.x - p1.x, -(p4.x - p3.x) * (p2.y - p1.y));
-    if d == 0.0 {
-        return None;
-    }
-    let u_a = (p4.x - p3.x).mul_add(p1.y - p3.y, -(p4.y - p3.y) * (p1.x - p3.x)) / d;
-    let u_b = (p2.x - p1.x).mul_add(p1.y - p3.y, -(p2.y - p1.y) * (p1.x - p3.x)) / d;
-    if matches!(bounds, IntersectionBounds::Unbounded)
-        || ((0.0..=1.0).contains(&u_a) && (0.0..=1.0).contains(&u_b))
-    {
-        return Some(Point::new(
-            u_a.mul_add(p2.x - p1.x, p1.x),
-            u_a.mul_add(p2.y - p1.y, p1.y),
-        ));
-    }
-    None
-}
-
-/// Determines whether the two line segments intersect.
-#[must_use]
-pub fn segments_intersect(line1: &Segment, line2: &Segment) -> bool {
-    intersection_of_lines(line1, line2, IntersectionBounds::Bounded).is_some()
-}
-
-/// Determines whether a line segment intersects a rectangle.
-#[must_use]
-pub fn rect_intersects_line(rect: &Rect, line: &Segment) -> bool {
-    let top_left = Point::new(rect.left() as SubPixelUnit, rect.top() as SubPixelUnit);
-    let top_right = Point::new(rect.right() as SubPixelUnit, rect.top() as SubPixelUnit);
-    let bottom_left = Point::new(rect.left() as SubPixelUnit, rect.bottom() as SubPixelUnit);
-    let bottom_right = Point::new(rect.right() as SubPixelUnit, rect.bottom() as SubPixelUnit);
-    let top_line = Segment::new(top_left, top_right);
-    let right_line = Segment::new(top_right, bottom_right);
-    let bottom_line = Segment::new(bottom_left, bottom_right);
-    let left_line = Segment::new(top_left, bottom_left);
-
-    segments_intersect(&top_line, line)
-        || segments_intersect(&bottom_line, line)
-        || segments_intersect(&left_line, line)
-        || segments_intersect(&right_line, line)
-}
-
 /// Returns the angle between two angles in radians.
 #[must_use]
 pub fn angle_diff(a: Radians, b: Radians) -> Radians {
-    let diff = normalize_angle(a - b);
-    diff.min(PI - diff)
+    let diff = normalize_angle(Radians::new(a.0 - b.0));
+    Radians::new(f32::min(diff.0, PI - diff.0))
 }
 
 /// Normalize angle to [0, PI). This means that two angles that are
 /// equivalent modulo PI will be equal, e.g. 90° and 270°, even though
 /// they are not equal in the mathematical sense.
 #[must_use]
 pub fn normalize_angle(angle: Radians) -> Radians {
```

Comment:
Do we still need this function if it's a method on `Radians` now?

Reply:
Nope, good catch. I'd removed this in an earlier iteration but it came back when I re-built it in commits.